R/create_covariates.R

```diff
@@ -1,217 +1,189 @@
+  if (!is.null(saga_path)) {
+    check_saga(saga_path)
+    cli::cli_alert_success(
+      "Your SAGA connection has been succesfuly set up. Using the {.var {saga_path}}"
+    )
   } else {
-    saga_cmd <- "saga_cmd"
-    fns      <- "/" ### file name separator
-
-  }  ;
-  z <- system(paste(saga_cmd, "-v"), intern = TRUE)  ## prints that SAGA version number -- confirming it works.
-  z <- print(z[1])
-  v <- suppressWarnings(as.numeric(unlist(strsplit(z, "[[:punct:][:space:]]+")[1])))
-  v <- v[!is.na(v)][1:2]
-  v <- as.numeric(paste(v[1], v[2], sep = "."))
-
-  if (v < 7.6) {
-    warning("SAGA-GIS is less that 7.6.  Not all covariates will generate.  Upgrade your SAGA, visit https://sourceforge.net/projects/saga-gis/files/")
+    check_saga()
+    cli::cli_abort("{.var saga_path} must point to the saga_path location on your computer.
+                   Please check your program files or equivalent to locate the saga_path.exe file")
   }
 
+
   #--- out_dir ---#
 
   if (!inherits(out_dir, c("character"))) {
     cli::cli_warn("{.var out_dir} must be  path to a file")
   }
 
 
-  #--- layers ---# have not updated this section yet
-
-  # layer_options <- c(
-  #   "sinksfilled", "sinkroute", "dem_preproc", "slope_aspect_curve",
-  #   "tcatchment", "tca", "scatchment", "twi", "channelsnetwork",
-  #   "overlandflow", "overlandflow2", "multiresflatness", "multiresflatness2",
-  #   "multiresflatness3", "tri", "convergence", "openness",
-  #   "dah", "tpi", "ridgevalley", "mrn", "flowaccumulation",
-  #   "slopelength", "flowaccumulation2", "flowaccumulation3",
-  #   "flowpathlength", "flowpathlength2", "flowpathlength3", "lsfactor",
-  #   "solarrad", "convexity", "vertdistance", "tci_low",
-  #   "swi", "windexp", "texture", "protection", "vrm",
-  #   "mbi", "mscale_tpi", "relposition", "slopecurvatures",
-  #   "steepestslope", "upslopearea"
-  # )
-  #
-  #
-
-  #--- layers ---# have not updated this section yet
+  #--- layers ---#
 
-  if(!is.character(layers)){
+  layer_options <- c(
+    "sinksfilled", "sinkroute", "dem_preproc", "slope_aspect_curve",
+    "tcatchment", "tca", "scatchment", "twi", "channelsnetwork",
+    "overlandflow", "overlandflow2", "multiresflatness", "multiresflatness2",
+    "multiresflatness3", "tri", "convergence", "openness",
+    "dah", "tpi", "ridgevalley", "mrn", "flowaccumulation",
+    "slopelength", "flowaccumulation2", "flowaccumulation3",
+    "flowpathlength", "flowpathlength2", "flowpathlength3", "lsfactor",
+    "solarrad", "convexity", "vertdistance", "tci_low",
+    "swi", "windexp", "texture", "protection", "vrm",
+    "mbi", "mscale_tpi", "relposition", "slopecurvatures",
+    "steepestslope", "upslopearea"
+  )
 
-    stop(paste0("`layers` must be type character."), call. = FALSE)
 
+  if (!inherits(layers, "character")) {
+    cli::cli_abort("{.var layers} must be a character string of the names of variables")
   }
 
   if (isTRUE(layers == "all")) {
     layers <- layer_options
   }
```

Comment:
```suggestion
  if (isTRUE(layers == "all")) {
    layers <- layer_options
  } else {
    layers <- c("dem_preproc", layers)
  }
```

Reply:
This would be adding dem_preproc, as it seems to be an unstated prerequisite for many